### PR TITLE
[dev] psycopg2: add doc subpackage

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Adds a doc subpackage for psycopg2

###### Does this affect the toolchain?  

NO

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- local build

### CODE REVIEW

File: SPECS/python-psycopg2/python-psycopg2.spec [hunk 1]
```diff
@@ -43,6 +43,15 @@ Requires:       postgresql >= 10.5
 %description -n python3-psycopg2
 Python 3 version.
 
+%package doc
+Summary:	Documentation for psycopg python PostgreSQL database adapter
+%{?with_python2:Provides: python2-%{srcname}-doc = %{version}-%{release}}
```
Comment (PawelWMS) [inline SPECS/python-psycopg2/python-psycopg2.spec:48]: We don't use the `with_X` macros too extensively. Please just keep the `Provides` for Python 2 and remove the macro logic.

File: SPECS/python-psycopg2/python-psycopg2.spec [hunk 2]
```diff
@@ -43,6 +43,15 @@ Requires:       postgresql >= 10.5
 %description -n python3-psycopg2
 Python 3 version.
 
+%package doc
+Summary:	Documentation for psycopg python PostgreSQL database adapter
```
Comment (PawelWMS) [inline SPECS/python-psycopg2/python-psycopg2.spec:47]: Please use spaces instead of tabs.

File: SPECS/python-psycopg2/python-psycopg2.spec [hunk 3]
```diff
@@ -95,7 +104,14 @@ rm -r /home/postgres/data &>/dev/null ||:
 %defattr(-,root,root,-)
 %{python3_sitelib}/*
 
+%files doc
+%license LICENSE
+%doc doc
+
 %changelog
+* Tue Jul 13 2021 Muhammad Falak Wani <mwani@microsoft.com> - 2.7.5-6
+- Enable subpackage python-psycopg2-doc
```
Comment (PawelWMS) [inline SPECS/python-psycopg2/python-psycopg2.spec:113]: Please also mention where we got the new spec parts from - same comment as in #1126.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/python-psycopg2/python-psycopg2.spec (hunk 1 above)
Comment (mfrw) [reply SPECS/python-psycopg2/python-psycopg2.spec:48]: Sure

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**